### PR TITLE
71-RoomFinder

### PR DESCRIPTION
**Pull request by ZYancey**
Finish findRoom functionality

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,6 +84,11 @@ Pulls the latest changes from the repository and restarts the bot.
 
 Begins a new game of Evil Hangman.
 
+
+### /findroom ( now / at / between / when )
+
+Searches for open rooms on BYU Campus. *'Now'* *'At'* and *'Between'* allow you to filter your search by time. *'When'* allows you to see when a specified room is available.
```
Comment (AverageHelper) [inline README.md:90]:
```suggestion
Searches for open rooms on BYU Campus. *'Now'*, *'At'*, and *'Between'* allow you to filter your search by time. *'When'* allows you to see when a specified room is available.
```

File: src/commands/findRoom.ts [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+	busySince: string;
+	busyUntil: string;
+	isInUse: boolean;
+}
+
+interface GetRoomsResponse {
+	Rooms: Array<[string, string]> | [];
+}
+
+function convertTo12Hour(time: string): string {
```
Comment (AverageHelper) [inline src/commands/findRoom.ts:16]: I think this function (and perhaps its neighbors) deserves a unit test or two

File: src/commands/findRoom.ts [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+						{ name: '7:30 PM', value: '19:30:00' }
+					)
+					.setRequired(true)
+			)
+			.addStringOption(option =>
+				option
+					.setName('building')
+					.setDescription('The building acronym to search in, type any to see all options')
+					.addChoices(
+						{ name: 'ANY', value: 'ANY' },
```
Comment (AverageHelper) [inline src/commands/findRoom.ts:190]: Might be wise to break this list of buildings into a string array, then construct these `name`/`value` object arrays in the four places they're needed

File: src/commands/findRoom.ts [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+	.addSubcommand(subcommand =>
+		subcommand
+			.setName('at')
+			.setDescription('Finds you an available room on campus at a given time!')
+			.addStringOption(option =>
+				option
+					.setName('start_time')
+					.setDescription('What time would you like to search for?')
+					.addChoices(
+						{ name: '8:00 AM', value: '08:00:00' },
```
Comment (AverageHelper) [inline src/commands/findRoom.ts:158]: Might be wise to break this list of times into an array, then re-use that array in the places it's needed

**Review by AverageHelper — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -84,6 +84,10 @@ Pulls the latest changes from the repository and restarts the bot.
 
 Begins a new game of Evil Hangman.
 
+
+### /findroom ( now / at / between / when )
+
+Searches for open rooms on BYU Campus. *'Now'*, *'At'*, and *'Between'* allow you to filter your search by time. *'When'* allows you to see when a specified room is available.
```
Comment (AverageHelper) [inline README.md:90]: Fixing spacing as per Prettier suggestion:

```suggestion
### /findroom ( now / at / between / when )

Searches for open rooms on BYU Campus. *'Now'*, *'At'*, and *'Between'* allow you to filter your search by time. *'When'* allows you to see when a specified room is available.

```

**Review by AverageHelper — DISMISSED** [review]

**Comment by AverageHelper** [comment]
The red arrows in the coverage report make me sad, lol

File: src/commands/findRoom.ts [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+	if (hours !== undefined && minutes !== undefined && seconds !== undefined) {
+		const suffix = hours >= 12 ? 'PM' : 'AM';
+		const convertedHours = hours % 12 || 12;
+		return `${convertedHours}:${String(minutes).padStart(2, '0')} ${suffix}`;
+	}
+	return 'ERR';
+}
+
+async function _getRoomsNow(search_type: string, building: string): Promise<GetRoomsResponse> {
+	try {
```
Comment (Plyb) [inline src/commands/findRoom.ts:82]: seems like we could collapse some of this by making a function that does all of this error handling

File: src/commands/findRoom.ts [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+		const input_timeB = interaction.options.getString('end_time');
+		const type = interaction.options.getSubcommand();
+
+		let embedTitle = '';
+		let embedDescription = '';
+		let embedThumbnail = 'https://pi.zyancey.com/img/room-finder-logo.png'; // not currently used, placeholder so that it doesn't show
+
+		if (type === 'now') {
+			if (input_bldg !== null) {
+				const requestedComic = await _getRoomsNow('now', input_bldg);
```
Comment (Plyb) [inline src/commands/findRoom.ts:259]: `comic` probably left over from the xkcd command

File: src/commands/findRoom.ts [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+		logger.error(error_);
+		const error: GetRoomsResponse = {
+			Rooms: [],
+		};
+		return error;
+	}
+}
+
+async function _getRoomsAt(
+	search_type: string,
```
Comment (Plyb) [inline src/commands/findRoom.ts:99]: It seems to me like `search_type` should probably just be a const inside this function (ditto for the other `getRooms` functions)

**Review by AverageHelper — APPROVED** [review]

**Review by Plyb — APPROVED** [review]

**Pull request merged**